rust/core-lib/src/editor.rs

```diff
@@ -649,17 +649,36 @@ impl Editor {
 
     fn insert_tab(&mut self) {
         let mut builder = delta::Builder::new(self.text.len());
+        let const_tab_text = if self.config.items.translate_tabs_to_spaces {
+                                    let tab_size = self.config.items.tab_size;
+                                    n_spaces(tab_size)
+                               } else { "\t" };
+
         for region in self.view.sel_regions() {
-            let iv = Interval::new_closed_open(region.min(), region.max());
-            let tab_text = if self.config.items.translate_tabs_to_spaces {
-                    let (_, col) = self.view.offset_to_line_col(&self.text, region.start);
-                    let tab_size = self.config.items.tab_size;
-                    let n = tab_size - (col % tab_size);
-                    n_spaces(n)
-            } else {
-                "\t"
-            };
-            builder.replace(iv, Rope::from(tab_text));
+            let mut sel_text = self.text.slice(region.min(), region.max());
```

Comment:
It would be more efficient to measure the text at the endpoints and compute the difference, than to allocate the slice.